### PR TITLE
Bounce, the real one

### PR DESCRIPTION
**Pull request by pablo-mayrgundter**
Same demo link, but now right PR..

https://pablo-mayrgundter.github.io/Share/84/103/138/160/93485

Recap: I sent #85, but it was actually just the merge of upstream main into my fork main.. no actual changes.  I thought I was in this branch but I wasn't.  All good now I hope!

### CODE REVIEW

File: public/404.html [hunk 1]
```diff
@@ -1 +1,39 @@
-<html><head><meta http-equiv="refresh" content="0; URL=/"></head><body></body></html>
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <title>Single Page Apps for GitHub Pages</title>
+    <script type="text/javascript">
+      // Single Page Apps for GitHub Pages
```
Comment (OlegMoshkovich) [inline public/404.html:7]: Do we need to include this description? @pablo-mayrgundter

File: public/index.html [hunk 2]
```diff
@@ -38,10 +38,38 @@
...
+          }).join('?');
+          var finalPath = l.pathname.slice(0, -1) + decoded + l.hash
+          window.history.replaceState(null, null, finalPath);
+        }
+      }(window.location))
+    </script>
+    <!-- End Single Page Apps for GitHub Pages -->
   </head>
   <body>
     <noscript>You need to enable JavaScript to run this app.</noscript>
```
Comment (OlegMoshkovich) [inline public/index.html:66]: minor: maybe we delete this?

File: public/index.html [hunk 3]
```diff
@@ -38,10 +38,38 @@
...
     <div id="root"></div>
-    <script src="./index.js"></script>
+    <script>
+      const loader = document.createElement('script');
+      const pathPrefix = window.location.pathname.startsWith('/Share') ? '/Share' : '';
```
Comment (OlegMoshkovich) [inline public/index.html:70]: nice

File: public/index.html (hunk 2 above)
Comment (pablo-mayrgundter) [reply public/index.html:66]: I think we leave in case some env doesn't have JS enabled, it'll otherwise fail silently.

Comment (OlegMoshkovich) [reply public/index.html:66]: ok

File: public/404.html (hunk 1 above)
Comment (pablo-mayrgundter) [reply public/404.html:7]: Yeah, it was meant as a copy/paste block from source, so keeping it intact is a nice nod to the original dev.  Also, makes it easy to compare with original later.

**Comment by pablo-mayrgundter** [comment]
Ok, trimmed some of the 404 code based on your comments. Still includes the link back to the proj.

File: public/404.html (hunk 1 above)
Comment (pablo-mayrgundter) [reply public/404.html:8]: Ok, did some trimming.  Please take another look (first PTAL!).

**Pull request merged**